### PR TITLE
Restore the ability to init NavVC without animations.

### PR DESCRIPTION
**Pull request by michaelkirk**
Prior to #54, the NavigationViewController could only be used modally.

With #54, you can switch between "map mode" and "navigation mode". 

An animation was introduced in `navigationView.showUI` for the sake of smoothly transitioning from "map mode" to "navigation mode".

However, if you are still using the NavigationViewController modally, this animation might interfere with other things you were doing - like zooming before presentation.

### CODE REVIEW

File: MapboxNavigation/NavigationViewController.swift [hunk 1]
```diff
@@ -496,8 +496,8 @@ open class NavigationViewController: UIViewController {
                 let directions = routeController.directions
                 let route = routeController.routeProgress.route
                 let navigationViewController = NavigationViewController(dayStyle: DayStyle(demoStyle: ()), nightStyle: NightStyle(demoStyle: ()), directions: directions)
-                navigationViewController.startNavigation(with: route, routeController: routeController, locationManager: locationManager)
-                
+                navigationViewController.startNavigation(with: route, animated: true, routeController: routeController, locationManager: locationManager)
```
Comment (hactar) [inline MapboxNavigation/NavigationViewController.swift:499]: Why do we pass animated: true in here instead of animated: animated?

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,19 +2,14 @@
...
-    if Env.current.simulateLocationForTesting {
-        let simulatedLocationManager = SimulatedLocationManager(route: route)
-        simulatedLocationManager.speedMultiplier = 5
-        vc.startNavigation(with: route, locationManager: simulatedLocationManager)
-    } else {
-        vc.startNavigation(with: route)
-    }
+    let simulatedLocationManager = SimulatedLocationManager(route: route)
+    simulatedLocationManager.speedMultiplier = 5
+    vc.startNavigation(with: route, animated: false, locationManager: simulatedLocationManager)
```
Comment (Patrick-Kladek) [inline CHANGELOG.md:12]: Why did you remove this? This way navigation will always simulate the route, which is helpful while working in the simulator but the opposite of what users want. It's not intuitive to omit this parameter (when the changelog says to pass it) to get default behaviour back. Maybe a comment here would be better suited

File: MapboxNavigation/NavigationViewController.swift [hunk 3]
```diff
@@ -422,7 +422,7 @@ open class NavigationViewController: UIViewController {
     
     // MARK: - NavigationViewController
 	
-    public func startNavigation(with route: Route, routeController: RouteController? = nil, locationManager: NavigationLocationManager? = nil) {
+    public func startNavigation(with route: Route, animated: Bool, routeController: RouteController? = nil, locationManager: NavigationLocationManager? = nil) {
```
Comment (Patrick-Kladek) [inline MapboxNavigation/NavigationViewController.swift:425]:
```suggestion
    public func startNavigation(with route: Route, animated: Bool = true, routeController: RouteController? = nil, locationManager: NavigationLocationManager? = nil) {
```

File: MapboxNavigation/NavigationViewController.swift (hunk 1 above)
Comment (Patrick-Kladek) [reply MapboxNavigation/NavigationViewController.swift:499]: This is called in 

```swift
func carPlayManager(_ carPlayManager: CarPlayManager, didBeginNavigationWith routeController: RouteController, window: UIWindow) 
```

It's hidden in github if you don't show more. I think passing animated: true is the right choice here.

**Review by Patrick-Kladek — APPROVED** [review]
The code looks good. I will approve it immediately; however, please address my comments.

File: CHANGELOG.md (hunk 2 above)
Comment (michaelkirk) [reply CHANGELOG.md:12]: I thought this was only supposed to be an example for the immediately preceding bullet point, which was about simulating location. My mistake! I'll revert it.

File: MapboxNavigation/NavigationViewController.swift (hunk 3 above)
Comment (michaelkirk) [reply MapboxNavigation/NavigationViewController.swift:425]: I typically use default values only when people will almost always want that default value. 

To me this feels closer to 50/50, so I'd prefer to have people type it out rather than run the risk of people assuming it doesn't exist. For example, anyone using the navigation VC modally (e.g. any legacy users), will want this to be false.

Comment (Patrick-Kladek) [reply MapboxNavigation/NavigationViewController.swift:425]: Okay makes sense, then let's keep it without the default parameter 👍

**Review by Patrick-Kladek — APPROVED** [review]

**Review by hactar — APPROVED** [review]
